### PR TITLE
started adding AuthentiCT module

### LINKED ISSUE #2886: new module: AUTHENTICT/DEAM2CONT
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by scarlhoff**
## PR checklist

Closes #2886 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    label 'process_single'
+
+    // TODO nf-core: List required Conda package(s).
+    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
+    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
+    conda "bioconda::authentict=1.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/authentict:1.0.0--py311h9f5acd7_0':
+        'quay.io/biocontainers/authentict:1.0.0--py311h9f5acd7_0' }"
```
Comment (StephanePeyregne) [inline modules/nf-core/authentict/deam2cont/main.nf:12]:
```suggestion
        'https://depot.galaxyproject.org/singularity/authentict:1.0.0--py311h9f5acd7_1':
        'quay.io/biocontainers/authentict:1.0.0--py311h9f5acd7_1' }"
```

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def config_file = config ? "-c ${config}" : ""
+    def positions_file = positions ? "-p ${positions}" : ""
+  
+    """
+    AuthentiCT \\
+        deam2cont \\
+        $args \\
+        $config_file \\
+        $positions_file \\
+        $bam \\
```
Comment (StephanePeyregne) [inline modules/nf-core/authentict/deam2cont/main.nf:39]:
```suggestion
```

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def config_file = config ? "-c ${config}" : ""
+    def positions_file = positions ? "-p ${positions}" : ""
+  
+    """
+    AuthentiCT \\
```
Comment (StephanePeyregne) [inline modules/nf-core/authentict/deam2cont/main.nf:34]:
```suggestion
    samtools view $args $bam | AuthentiCT \\
```

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    output:
+    tuple val(meta), path("*.txt"), emit: txt
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
```
Comment (StephanePeyregne) [inline modules/nf-core/authentict/deam2cont/main.nf:27]:
```suggestion
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
```

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.0' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
+    def config_file = config ? "-c ${config}" : ""
+    def positions_file = positions ? "-p ${positions}" : ""
+  
+    """
+    AuthentiCT \\
+        deam2cont \\
+        $args \\
```
Comment (StephanePeyregne) [inline modules/nf-core/authentict/deam2cont/main.nf:36]:
```suggestion
        $args2 \\
```

**Review by StephanePeyregne — CHANGES_REQUESTED** [review]

**Comment by scarlhoff** [comment]
Samtools version in the container is broken, replace with a different version
remember MD fields required for input

File: modules/nf-core/authentict/deam2cont/main.nf [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        deam2cont \\
+        $args2 \\
+        $config_file \\
+        $positions_file \\
+        - \\
+        > ${prefix}.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        authentict: \$(echo \$(AuthentiCT --version 2>&1) )
```
Comment (jfy133) [inline modules/nf-core/authentict/deam2cont/main.nf:40]: Missing `samtools` version

File: modules/nf-core/authentict/deam2cont/meta.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      description: BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
+  - config:
+      type: file
+      description: AuthentiCT configuration text file
+      pattern: "*"
+  - positions:
+      type: file
+      description: text file with positions that sequences should overlap
+      pattern: "*"
```
Comment (jfy133) [inline modules/nf-core/authentict/deam2cont/meta.yml:36]: Missing indication whether the files are mandatory or optional

**Review by jfy133 — CHANGES_REQUESTED** [review]
Almost there :)

File: modules/nf-core/authentict/deam2cont/meta.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      type: file
+      description: BAM/CRAM/SAM file
+      pattern: "*.{bam,cram,sam}"
+  - config:
+      type: file
+      description: AuthentiCT configuration text file
+      pattern: "*"
+  - positions:
+      type: file
+      description: text file with positions that sequences should overlap
```
Comment (scarlhoff) [inline modules/nf-core/authentict/deam2cont/meta.yml:35]:
```suggestion
      description: Optional text file with positions that sequences should overlap
```

File: modules/nf-core/authentict/deam2cont/meta.yml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - bam:
+      type: file
+      description: BAM/CRAM/SAM file
```
Comment (jfy133) [inline modules/nf-core/authentict/deam2cont/meta.yml:27]:
```suggestion
      description: BAM/CRAM/SAM file (Mandatory)
```

**Review by jfy133 — APPROVED** [review]

**Review by StephanePeyregne — APPROVED** [review]

**Pull request merged**